### PR TITLE
Rewrite the file exclusion to use minimatch

### PR DESCRIPTION
**Pull request by minj**
Required some hacking to support extraction filter

### CODE REVIEW

**Comment by minj** [comment]
proposal to exclude these as well:
`**/*.{zip,xpi,crx,safariextz}`

File: libs/utils/unzip-crx.js [hunk 1]
```diff
@@ -4,20 +4,52 @@ var jszip = require('jszip');
...
+    // and our glob may match some directory up the tree
+    // without matching the file entry itself
+    var excluded = function (filename) {
+        if (typeof filter === 'function') {
+            // this short-circuits the filter function
+            // to prevent endless spam down the tree
+            if (prunedPaths.some(p => filename.startsWith(p)))
+                return true;
 
+            var parts = filename.split(/\//);
```
Comment (abarreir) [inline libs/utils/unzip-crx.js:22]: No need for `path.sep` here?

File: libs/utils/unzip-crx.js [hunk 2]
```diff
@@ -4,20 +4,52 @@ var jszip = require('jszip');
...
+            var partial = '';
+            for (var part of parts) {
+                if (partial.length === 0)
```
Comment (abarreir) [inline libs/utils/unzip-crx.js]: Please use accolades for consistence.

File: libs/utils/unzip-crx.js [hunk 3]
```diff
@@ -4,20 +4,52 @@ var jszip = require('jszip');
...
+                    return true;
+                }
+            }
+        }
+        return false;
+    };
+
+    var zip = new jszip(buffer);
     async.eachSeries(Object.keys(zip.files), function (filename, onDone) {
+        if (excluded(filename))
```
Comment (abarreir) [inline libs/utils/unzip-crx.js]: Please use accolades for consistence.

**Comment by abarreir** [comment]
Looking good! Not sure about your proposal though, sounds specific to your use case.

File: libs/utils/unzip-crx.js (hunk 1 above)
Comment (minj) [inline libs/utils/unzip-crx.js:22]: jszip seems to be using / exclusively

**Comment by minj** [comment]
I think the zip specification calls for `/` only

No references to sep here:
- https://github.com/Stuk/jszip/blob/master/lib/index.js
- https://github.com/Stuk/jszip/blob/master/lib/object.js

**Pull request merged**